server/src/entities/StudentTeacher.ts

```diff
@@ -0,0 +1,20 @@
+import { Entity, PrimaryKey, Property } from "mikro-orm";
+import { Field, ObjectType } from "type-graphql";
+
+@ObjectType()
+@Entity()
+export class StudentTeacher {
+
+  @Field()
+  @PrimaryKey()
+  id!: number;
+
+  @Field()
+  @Property({type: 'text'})
+  student!: string;
```

Comment:
You don't need an extra id. The student id should be the primary key (a student can only join one teacher's class)